src/goose/client.clj

```diff
@@ -156,4 +163,13 @@
 
   - [Periodic Jobs wiki](https://github.com/nilenso/goose/wiki/Periodic-Jobs)"
   [opts cron-opts execute-fn-sym & args]
-  (register-cron-schedule opts cron-opts execute-fn-sym args))
+  (let [internal-opts (prefix-queues-inside-opts opts)]
+    (register-cron-schedule internal-opts cron-opts execute-fn-sym args)))
+
+(defn perform-batch
+  "Enqueues a batch of jobs for async execution."
+  ([opts batch-opts execute-fn-sym args-coll]
+   (let [internal-opts (prefix-queues-inside-opts opts)
+         jobs (map #(create-job internal-opts execute-fn-sym %) args-coll)
+         batch (batch/new internal-opts batch-opts jobs)]
+     (b/enqueue-batch (:broker internal-opts) batch))))
```

Comment:
we are passing the `internal opts` twice? first in `(:broker internal-opts)` and second in the `batch`